pkg/utils/secrets/infodata/basic_auth.go

```diff
@@ -0,0 +1,68 @@
+// Copyright (c) 2020 SAP SE or an SAP affiliate company. All rights reserved. This file is licensed under the Apache Software License, v. 2 except as noted otherwise in the LICENSE file
```

Comment:
All these new files+functions of this `infodata` package seem to be tightly coupled to the `pkg/utils/secrets` package. Does it make sense to merge them into it?

Reply:
If we want to have the infodata only in Gardener then I guess I should try to make them more independent, as the secret configs are used in some extension controllers (however, i don't have any good ideas to that currently). Otherwise I could add them to the same package (simiar to the etcdencryption)

Reply:
I would agree with @rfranzke, the `infodata` package is too tightly coupled to deserve an own package and the split of the code between the 2 packages makes reading and understanding the code harder. So I would also vote for a single package (and set of files).